android/app/src/main/java/com/byu/project_apraxia/project_apraxia/MainActivity.java

```diff
@@ -0,0 +1,130 @@
+package com.byu.project_apraxia.project_apraxia;
+
+import android.os.Bundle;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import io.flutter.app.FlutterActivity;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+import io.flutter.plugins.GeneratedPluginRegistrant;
+
+public class MainActivity extends FlutterActivity {
+    private static final String CHANNEL = "wsdCalculator";
+    private Double threshold = 0.0040517578125;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        GeneratedPluginRegistrant.registerWith(this);
+
+        new MethodChannel(getFlutterView(), CHANNEL).setMethodCallHandler(
+                new MethodCallHandler() {
+                    @Override
+                    public void onMethodCall(MethodCall call, Result result) {
+                        // Note: this method is invoked on the main thread.
+                        switch (call.method) {
+                            case "calculateAmbiance":
+                                String ambianceFileName = (String) ((ArrayList) call.arguments()).get(0);
+
+                                try {
+                                    WavFile ambianceFile = WavFile.openWavFile(new File(ambianceFileName));
+                                    List<Double> ambianceAmplitudeList = getAmplitudeList(ambianceFile);
+                                    double threshold = getAmbianceThreshold(ambianceAmplitudeList);
```

Comment:
So I think here shouldn't be a new double? It should just reference the class's threshold variable